MetaMask/metamask-extension
# feat: Increase token symbol maximum character length from 11 to 64

## **Description**

Increase the token symbol length to 64.

With the increase of token derivatives, token symbol lengths are also increasing.

When adding tokens to the wallet by RPC call, Metamask expects the token symbol to match that on the contract and the symbol to be below 12 characters.

For many tokens (like tenderize's LSTs) this results in a deadlock if the token symbol on the contract has a greater length than 11.

Since Metamask already truncates display of long symbols on the UI, we could allow token symbols up to 64 characters (32 bytes), which equals a single word in the EVM.

This should both be a sufficient length for 99% of token symbols as well as provide a sufficient upper bound.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27175?quickstart=1)

## **Related issues**

Fixes:
- https://github.com/MetaMask/metamask-extension/issues/9243
- https://github.com/MetaMask/metamask-extension/issues/11453

Same implementation as: https://github.com/MetaMask/metamask-extension/pull/5816

## **Manual testing steps**

1. Go to import a token.
2. Enter a contract address of a token with a name longer than 11 characters (the previous limit) and with a maximum of 64 characters (the new limit).
3. No error message should appear, and the token can be imported.

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
